iceoryx2/src/config.rs

```diff
@@ -354,6 +364,82 @@ impl Default for Config {
 }
 
 impl Config {
+    fn relative_local_config_path() -> Path {
+        fatal_panic!(from "Config::default_config_path",
+            when Path::new(RELATIVE_LOCAL_CONFIG_PATH),
+            "This should never happen! The default config path contains invalid symbols.")
+    }
+
+    /// The name of the default iceoryx2 config file
+    pub fn default_config_file_name() -> FileName {
+        fatal_panic!(from "Config::default_config_file",
+            when FileName::new(DEFAULT_CONFIG_FILE_NAME),
+            "This should never happen! The default config file name contains invalid symbols.")
+    }
+
+    /// Path to the default config file
+    pub fn default_config_file_path() -> FilePath {
+        fatal_panic!(from "Config::default_config_file_path",
+            when FilePath::from_path_and_file(&Self::relative_local_config_path(), &Self::default_config_file_name()),
+            "This should never happen! The default config file path contains invalid symbols.")
+    }
+
+    fn relative_config_path() -> Path {
+        fatal_panic!(from "Config::relative_config_path",
+            when Path::new(RELATIVE_CONFIG_FILE_PATH),
+            "This should never happen! The relative config path contains invalid symbols.")
+    }
+
+    fn iterate_over_config_files<F: FnMut(FilePath) -> CallbackProgression>(
+        mut callback: F,
+    ) -> Result<(), ConfigIterationFailure> {
```

Comment:
What do you think about changing this to
```rs
fn get_config_file() -> Result<Option<Path>, Error> {}
```
On error, the caller can panic and if it was successful, the caller can try to open the config or use the default config if no path was found.

This would simplify the code a bit.

Reply:
But this function iterates over multiple config files? So one option wouldn't be sufficient.

Reply:
No, it requires one config file to be present at one of these locations. Once the config file is present it is either valid or the caller panics.

Reply:
Aaah understood!